homeassistant/components/mikrotik/config_flow.py

```diff
@@ -0,0 +1,134 @@
+"""Config flow for Mikrotik."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+
+from .const import (
+    CONF_ARP_PING,
+    CONF_DETECTION_TIME,
+    CONF_TRACK_DEVICES,
+    DEFAULT_API_PORT,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+from .errors import CannotConnect, LoginError
+from .hub import get_hub
+
+
+class MikrotikFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Mikrotik config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return MikrotikOptionsFlowHandler(config_entry)
+
+    def __init__(self):
+        """Initialize the UniFi flow."""
+        self.config = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    return self.async_abort(reason="already_configured")
+
+            errors = self.validate_user_input(user_input)
+            if not errors:
+                return self.async_create_entry(
+                    title=self.config[CONF_NAME], data=self.config
+                )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(CONF_PORT, default=DEFAULT_API_PORT): int,
+                    vol.Optional(CONF_VERIFY_SSL, default=False): bool,
+                    vol.Optional(CONF_TRACK_DEVICES, default=False): bool,
```

Comment:
We need to check with @balloob if it's ok to have this option in the config flow.

Reply:
@balloob What is your recommendation for this?